src/schema/v2/me/updateUserInterestMutation.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  GraphQLBoolean,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GravityMutationErrorType } from "lib/gravityErrorHandler"
+import { ResolverContext } from "types/graphql"
+import { UserInterest, userInterestType } from "../userInterests"
+
+interface Input {
+  id: string
+  private?: boolean
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateUserInterestSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    userInterest: {
+      type: userInterestType,
+      resolve: async (response) => {
+        return response
+      },
+    },
+  }),
+})
+
+const ErrorType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateUserInterestFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => (typeof err.message === "object" ? err.message : err),
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "UpdateUserInterestResponseOrError",
+  types: [SuccessType, ErrorType],
+})
+
+export const updateUserInterestMutation = mutationWithClientMutationId<
+  Input,
+  UserInterest | null,
+  ResolverContext
+>({
+  name: "UpdateUserInterestMutation",
+  description:
+    "Updates a UserInterest on the logged in User's CollectorProfile.",
+  inputFields: {
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    private: { type: GraphQLBoolean },
+  },
+  outputFields: {
+    userInterestOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: the new state of the UserInterest",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (
+    { id, private: isPrivate },
+    { meUpdateUserInterestLoader }
+  ) => {
+    if (!meUpdateUserInterestLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    try {
+      const userInterest: UserInterest = await meUpdateUserInterestLoader?.(
+        id,
+        { private: !!isPrivate }
```

Comment:
**question**: Let's imaging that in future this mutation will support other input fields and `private` field is not sent as an input argument (something like this):

```graphql
mutation {
  updateUserInterest(
      input: { id: "id", someOtherField: "newValue" }
   ) {
    ...
  }
}
```

If I understand this code correctly, then `private` will be sent to gravity in any case as `false`, am I right? It might be not very transparent for developers why `private` suddenly becomes `false`. I think that we should either handle this in the code (if field is not passed - do not send it to gravity) or at least make this field required)

@MounirDhahri

Reply:
good catch 👍

Reply:
will remove the `!!`